src/evtx_parser.rs

```diff
@@ -349,12 +350,12 @@ impl<T: ReadSeek> EvtxParser<T> {
             current_chunk_number: 0,
         }
     }
-
     /// Return an iterator over all the records.
-    /// Records will be serialized using the given `BinXmlOutput`.
-    pub fn serialized_records<O: BinXmlOutput<Vec<u8>>>(
-        &mut self,
-    ) -> impl Iterator<Item = Result<SerializedEvtxRecord>> + '_ {
+    /// Records will be mapped `f`, which must produce owned data from the records.
+    pub fn owned_records<'a, U: Send>(
+        &'a mut self,
+        f: impl FnMut(Result<EvtxRecord<'_>>) -> Result<U> + Send + Sync + Copy + 'a,
```

Comment:
Why is the copy bound needed?

Reply:
I'll let you hear it from the horse's mouth:
```
error[E0507]: cannot move out of `f`, a captured variable in an `Fn` closure
...
393 | Ok(mut chunk_records) => chunk_records.iter().map(f).collect(),
    |                                                   ^ move occurs because `f` has type `impl FnMut(Result<EvtxRecord<'_>>) -> Result<U> + Send + Sync + 'a`, which does not implement the `Copy` trait
```

(TLDR: We reuse the closure for every chunk, and `map` require ownership on the provided clouser)